installers/msi-language/GAPixel/GAPixelCA.cs

```diff
@@ -0,0 +1,256 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Deployment.WindowsInstaller;
+using System.Management;
+using System.Net;
+using Microsoft.Win32;
+
+namespace GAPixel
+{
+    public class GetInfo
+    {
+
+        private static Guid StringToGuid(string input)
+        {
+            using (System.Security.Cryptography.MD5 md5 = System.Security.Cryptography.MD5.Create())
+            {
+                byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(input);
+                byte[] hashBytes = md5.ComputeHash(inputBytes);
+
+                return new Guid(hashBytes);
+            }
+        }
+
+        public static string GetUniqueId(Session session)
+        {
+            try
+            {
+                var oMClass = new ManagementClass("Win32_NetworkAdapterConfiguration");
+                var colMObj = oMClass.GetInstances();
+                foreach (var objMO in colMObj)
+                {
+                    try
+                    {
+                        var macAddress = objMO["MacAddress"].ToString();
+                        if (String.IsNullOrEmpty(macAddress))
+                        {
+                            continue;
+                        }
+                        // return on first found MAC address
+                        session.Log(String.Format("Found MAC={0}", macAddress));
+                        return StringToGuid(macAddress.ToString()).ToString();
+                    } catch(NullReferenceException)
+                    {
+                        continue;
+                    }
+                }
+            }
+            catch (Exception err)
+            {
+                session.Log(String.Format("Error getting unique ID {0}", err));
+            }
+            // fallback method
+            return Guid.NewGuid().ToString();
```

Comment:
Can we use something static as a fallback? That way we can identify if this is in fact not working.